lib/mempool/addrindexer.js

```diff
@@ -0,0 +1,235 @@
+/*!
+ * mempool.js - mempool for bcoin
+ * Copyright (c) 2018-2019, the bcoin developers (MIT License).
+ * Copyright (c) 2014-2017, Christopher Jeffrey (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const {BufferMap} = require('buffer-map');
+const TXMeta = require('../primitives/txmeta');
+
+/**
+ * Address Indexer
+ * @ignore
+ */
+
+class AddrIndexer {
+  /**
+   * Create TX address index.
+   * @constructor
+   */
+
+  constructor() {
+    // Map of addr->entries.
+    this.index = new BufferMap();
+
+    // Map of txid->addrs.
+    this.map = new BufferMap();
+  }
+
+  reset() {
+    this.index.clear();
+    this.map.clear();
+  }
+
+  getKey(addr) {
+    const prefix = addr.getPrefix();
```

Comment:
If understand this prefix is not crucial to the mempool, I noticed that you did not pass `getPrefix(this.network)` so this is not passing network is not necessary.

Reply:
The prefix resolved https://github.com/bcoin-org/bcoin/issues/738, it's also used in the addrindexer. I don't think having the prefix for the network would change the behavior, nor harm it not being of the correct network. It could be PUBKEYHASH = 0, SCRIPTHASH = 1, WITNESS_PUBKEYHASH = 3, and WITNESS_SCRIPTHASH = 4 for example.

Reply:
There could be potential issues with the primary global network being changed, so I think it's best to pass the network. Sounds good.